cmd/image-processing/main.go

```diff
@@ -179,6 +187,44 @@ func runImageProcessing(ctx context.Context) error {
 		}
 	}
 
+	// check for image vulnerabilities if vulnerability scanning is enabled.
+	var vulns []buildapi.Vulnerability
+	if flagValues.vulnerabilitySettings.Enabled {
+		var imageString string
+		if flagValues.push != "" {
+			imageString = flagValues.push
+			// for single image in a tar file
+			if img != nil {
+				entries, err := os.ReadDir(flagValues.push)
+				if err != nil {
+					return err
+				}
+				imageString = filepath.Join(imageString, entries[0].Name())
+			}
+			auth = nil
+			flagValues.insecure = false
+		} else {
+			imageString = imageName.String()
+		}
+		vulns, err = image.RunVulnerabilityScan(ctx, imageString, flagValues.vulnerabilitySettings.VulnerabilityScanOptions, auth, flagValues.insecure)
+		if err != nil {
+			return err
+		}
+		vulnsData, err := json.Marshal(vulns)
```

Comment:
We are writing a result here. The overall size of results is limitted by the maximum termination message size of a Pod, which is 4096 bytes. Because of this, we must be cautious in how much data we write. Because of this, I suggested the vulnerability limit. And, because of this, I also suggested a different serialization format in the ship:

> The serialization format is a comma-separated string (`vuln1-id:vuln1-severity,vuln2-id:vuln2-severity`) to save characters because the total number of result data is limited.

Using this is massively shorter than what you are doing. In above serialized format, you can even shorten severities to just l, m, h, c (for low, medium, high, critical) to save even further bytes.

You can read more about this here: https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md. Tekton in the meantime also supports larger results with sidecar logs, but Shipwright does not support that.